CMS-Enterprise/easi-app
# EASI-4649 Upgrade vite & vitest

# EASI-4649

## Description

- Bumps up `vite` and `vitest` to current
- Adds github actions reporter to client test

> [!NOTE]
> There was an attempt to change package.json `type` to `module`, but this was reverted and on hold because some dependencies and confs still rely on commonjs.

## How to test this change

Reinstall npm packages and expect business as usual.
```
yarn install
yarn test
yarn start
yarn cypress:run
```
## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
